### PR TITLE
docs(headless): export and document StaticFilter

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-1110

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]
Do we have plans for supporting this inside Atomic?

File: packages/headless/src/controllers/static-filter/headless-static-filter.ts [hunk 1]
```diff
@@ -65,6 +70,14 @@ export interface StaticFilter extends Controller {
    * */
   deselectAll(): void;
 
+  /**
+   * Checks whether the specified static filter value is selected.
+   *
+   * @param value - The static filter value to check.
+   * @returns Whether the specified static filter value is selected.
+   */
+  isValueSelected(value: StaticFilterValue): boolean;
```
Comment (btaillon) [inline packages/headless/src/controllers/static-filter/headless-static-filter.ts:79]: I don't know if this is necessary, especially if we take the object as a parameter.

File: packages/samples/headless-react/src/pages/SearchPage.tsx [hunk 2]
```diff
@@ -382,6 +385,21 @@ export class SearchPage extends Component {
...
+      {
+        caption: 'Youtube',
+        expression: '@filetype==youtubevideo',
+        state: 'idle',
+      },
+      {
+        caption: 'Dropbox',
+        expression: '(@connectortype==DropboxCrawler AND @objecttype==File)',
+        state: 'idle',
+      },
```
[the remark this answers is not in the document]
Comment (btaillon) [reply packages/samples/headless-react/src/pages/SearchPage.tsx:399]: I like that, I think we should provide it.

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/controllers/static-filter/headless-static-filter.ts (hunk 1 above)
Comment (olamothe) [reply packages/headless/src/controllers/static-filter/headless-static-filter.ts:79]: Yeah, that seems very superfluous as it currently stands.

Comment (samisayegh) [reply packages/headless/src/controllers/static-filter/headless-static-filter.ts:88]: I'm going to keep the method ... If you feel it should be removed, I can do a pass through the static filter and facet controllers and deprecate it in a separate PR.

**Pull request merged**